File: backend/app/schemas/household.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, EmailStr, Field

from app.models.enums import HouseholdMemberStatus, HouseholdRole


class HouseholdCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)


class HouseholdUpdate(BaseModel):
    name: Optional[str] = Field(default=None, min_length=1, max_length=255)


class HouseholdMemberRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    email: str
    full_name: Optional[str] = None
    role: HouseholdRole
    status: HouseholdMemberStatus
    created_at: datetime


class HouseholdRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    owner_id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    members: list[HouseholdMemberRead] = Field(default_factory=list)


class HouseholdInviteCreate(BaseModel):
    email: EmailStr


class PendingInvitationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    household_id: uuid.UUID
    household_name: str
    role: HouseholdRole
    created_at: datetime

File: frontend/src/types/household.ts
export type HouseholdRole = "owner" | "member"
export type HouseholdMemberStatus = "pending" | "accepted"

export interface HouseholdMember {
  id: string
  user_id: string
  email: string
  full_name: string | null
  role: HouseholdRole
  status: HouseholdMemberStatus
  created_at: string
}

export interface Household {
  id: string
  name: string
  owner_id: string
  created_at: string
  updated_at: string
  members: HouseholdMember[]
}

export interface HouseholdPayload {
  name?: string
}

export interface InviteMemberPayload {
  email: string
}

export interface PendingInvitation {
  id: string
  household_id: string
  household_name: string
  role: HouseholdRole
  created_at: string
}
